src/test/java/teammates/test/cases/browsertests/InstructorCourseDetailsPageUiTest.java

```diff
@@ -269,12 +270,37 @@ private InstructorCourseDetailsPage getCourseDetailsPage() {
 
     private boolean hasStudentReceivedReminder(String courseName, String courseId, String studentEmail)
             throws Exception {
-        String keyToSend = BackDoor.getEncryptedKeyForStudent(courseId, studentEmail);
 
-        ThreadHelper.waitFor(5000); //TODO: remove this and use RetryManager with a shorter wait interval
-        String keyReceivedInEmail =
-                EmailAccount.getRegistrationKeyFromGmail(studentEmail, courseName, courseId);
-        return keyToSend.equals(keyReceivedInEmail);
+        RetryManager retryManager = new RetryManager(3);
+
+        return retryManager.runUntilSuccessful(new RetryableTaskReturns<Boolean>("Received email") {
+
+            String keyToSend;
+            String keyReceivedInEmail;
+            boolean isDoneExecuting;
+
+            @Override
+            public Boolean run() throws RuntimeException {
+
+                keyToSend = BackDoor.getEncryptedKeyForStudent(courseId, studentEmail);
```

Comment:
Isn't this constant? Why is there a need to put this in `run()`?

Reply:
@LiHaoTan If not just last 5 then how many?

Reply:
Let's fetch everything?